src/main/scala/com/codacy/parsers/implementation/CloverParser.scala

```diff
@@ -0,0 +1,71 @@
+package com.codacy.parsers.implementation
+
+import java.io.File
+
+import com.codacy.api.{CoverageFileReport, CoverageReport}
+import com.codacy.parsers.CoverageParser
+import com.codacy.parsers.util.{TextUtils, XMLoader}
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{Elem, Node, NodeSeq}
+
+object CloverParser extends CoverageParser {
+  private val CoverageTag = "coverage"
+  private val ProjectTag = "project"
+  private val MetricsTag = "metrics"
+
+  override val name: String = "Clover"
+
+  override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] = {
+    val report = Try(XMLoader.loadFile(reportFile)) match {
+      case Success(xml) if hasCorrectSchema(xml) =>
+        Right(xml \\ CoverageTag)
+
+      case Success(_) =>
+        Left(s"Invalid report. Could not find tag hierarchy <$CoverageTag> <$ProjectTag> <$MetricsTag> tags.")
+
+      case Failure(ex) =>
+        Left(s"Unparseable report. ${ex.getMessage}")
+    }
+
+    report.right.map(parse(rootProject, _))
+  }
```

Comment:
Wrap the call to inner parse in a Try:
```suggestion
  override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] = {
    val report = Try(XMLoader.loadFile(reportFile)).map { xml =>
      if hasCorrectSchema(xml) 
        Right(xml \\ CoverageTag)
      else 
        Left(s"Invalid report. Could not find tag hierarchy <$CoverageTag> <$ProjectTag> <$MetricsTag> tags.")
    }
    report.right.flatMap { node =>
      Try(parse(rootProject, node)) match {
        case Success(x) => Right(x)
        case Failure(ex) => Left(s"Unparseable report. ${ex.getMessage}")
      }
    }
  }
```

Reply:
I added the Try around the inner parse, but left the Success(_) in the first try, since it can be used to provide a better error message